packages/google-closure-compiler/lib/grunt/index.js

```diff
@@ -194,29 +202,46 @@ module.exports = (grunt, pluginOptions) => {
         options.compilerOpts.js_output_file = f.dest;
       }
 
-      compileTasks.push(compilationPromise(src, options.args || options.compilerOpts, {platform})
-          .then(function () {}, function(err) {
-            throw err;
-          }));
+      compileTasks.push(compilationPromiseGenerator(src, options.args || options.compilerOpts, {platform}));
     });
 
     // If the task was invoked without any files provided by grunt, assume that
     // --js flags are present and we want to run the compiler anyway.
     if (taskObject.files.length === 0) {
       const options = getCompilerOptions();
-      compileTasks.push(compilationPromise(null, options.args || options.compilerOpts, {platform}));
+      compileTasks.push(compilationPromiseGenerator(null, options.args || options.compilerOpts, {platform}));
     }
 
     // Multiple invocations of the compiler can occur for a single task target. Wait until
     // they are all completed before calling the "done" method.
-    Promise.all(compileTasks)
+
+    return compileInBatches ? processPromises(compileTasks, asyncDone) : Promise.all(compileTasks.map(t => t()))
       .then(() => asyncDone())
       .catch(() => {
         grunt.fail.warn('Compilation error');
         asyncDone();
       });
   }
 
+  function processPromises(ps, done) {
```

Comment:
Can you add a JSDoc comment with an explanation of what this function is doing?

Reply:
Added inline comments as in other functions and into function body too. Is it good description?